sklearn/model_selection/tests/test_search.py

```diff
@@ -1885,3 +1887,27 @@ def _fit_param_callable():
         'scalar_param': 42,
     }
     model.fit(X_train, y_train, **fit_params)
+
+
+@pytest.mark.parametrize(
+    "Estimator,params,estimator_type",
+    [(DecisionTreeClassifier, {"max_depth": [2, 3]}, "classifier"),
+     (DecisionTreeRegressor, {"max_depth": [2, 3]}, "regressor"),
+     (KMeans, {"n_clusters": [2, 3]}, "clusterer"),
+     (BayesianGaussianMixture, {"n_components": [2, 3]}, "DensityEstimator"),
+     (IsolationForest, {"max_depth": [2, 3]}, "outlier_detector")]
+)
+@pytest.mark.parametrize("SearchCV", [GridSearchCV, RandomizedSearchCV])
+def test_estimator_type_tag(SearchCV, Estimator, params, estimator_type):
+    # Assert that estimator_type tag is properly set
+    search = SearchCV(Estimator(), params)
```

Comment:
Would this work? Since we don't fit anyway this could remove noise

```suggestion
    search = SearchCV(Estimator(), param_grid={})
```

Reply:
This does not work because the `sklearn.model_selection.RandomizedSearchCV` meta-estimator takes `param_distributions` instead of `param_grid`.

Nevertheless, the following works:

```suggestion
    search = SearchCV(Estimator(), {})
```

Thanks for the suggestion!